larq_compute_engine/tflite/tests/quantization_test.cc

```diff
@@ -0,0 +1,132 @@
+#include <random>
+
+#include "larq_compute_engine/core/bitpack_utils.h"
+#include "larq_compute_engine/core/types.h"
+#include "larq_compute_engine/tflite/tests/utils.h"
+#include "tensorflow/lite/kernels/test_util.h"
+
+using namespace tflite;
+using namespace compute_engine::core;
+
+namespace compute_engine {
+namespace tflite {
+
+TfLiteRegistration* Register_QUANTIZE();
+TfLiteRegistration* Register_DEQUANTIZE();
+
+namespace testing {
+
+typedef TfLiteRegistration* (*register_function)(void);
+
+class QuantizationOpModel : public SingleOpModel {
+ public:
+  QuantizationOpModel(register_function registration, const TensorData& input,
+                      const TensorData& output) {
+    input_ = AddInput(input);
+    output_ = AddOutput(output);
+    // It doesn't matter if the name is not correct
+    SetCustomOp("LceQuantization", {}, registration);
```

Comment:
I don't quite understand the comment here, why doesn't it matter?

Reply:
Normally, when a model is loaded, the `OpResolver` maps the op names to the corresponding registration functions. In this testing setting, however, we directly build an interpreter object with the model 'already loaded' and op names already resolved. The only effect of the string here, is that when the op throws an error it'll print a message like `"Node 0 (LceQuantization) failed to prepare."`. And since I'm using this class for both the Quantize and Dequantize op, I thought I'd just give it a single name that is a bit more general.